### PR TITLE
Improve plugin test util

### LINKED ISSUE #1349: [Test util] Remove vDVS plugin test util needs to be improved
As described in #1348, tearDown method needs to be improved after getting fix for #1348.

### PR DESCRIPTION
**Pull request by ashahi1**
Fixes #1349 
Util name: dockercli.DeleteVolume()

Now DeleteVolume util will check if deletion operation failed because plugin was still not completely initialized. If that's the case, we retry to delete volume. We keep trying for 30 seconds at an interval of two seconds and then fail the operation if volume is not successfully removed at the end of 30 seconds.

Ran the tests locally. Tests were successful.

Log output after change:
```
2017/06/08 18:36:14 Destroying volume [testVolXXXXX]
2017/06/08 18:36:54 Failed to invoke command [docker volume rm testVolXXXXX]: exit status 1
2017/06/08 18:36:54 Sleep for 2 seconds
2017/06/08 18:36:56 Volume cannot be deleted yet as plugin initialization still in progress. Retrying...
2017/06/08 18:36:57 Failed to invoke command [docker volume rm testVolXXXXX]: exit status 1
2017/06/08 18:36:57 Sleep for 2 seconds
2017/06/08 18:36:59 Volume cannot be deleted yet as plugin initialization still in progress. Retrying...
2017/06/08 18:37:00 Failed to invoke command [docker volume rm testVolXXXXX]: exit status 1
2017/06/08 18:37:00 Sleep for 2 seconds
2017/06/08 18:37:02 Volume cannot be deleted yet as plugin initialization still in progress. Retrying...
2017/06/08 18:37:02 Failed to invoke command [docker volume rm testVolXXXXX]: exit status 1
2017/06/08 18:37:02 Sleep for 2 seconds
2017/06/08 18:37:04 Volume cannot be deleted yet as plugin initialization still in progress. Retrying...
2017/06/08 18:37:05 Failed to invoke command [docker volume rm testVolXXXXX]: exit status 1
2017/06/08 18:37:05 Sleep for 2 seconds
2017/06/08 18:37:07 Volume cannot be deleted yet as plugin initialization still in progress. Retrying...
2017/06/08 18:37:08 Failed to invoke command [docker volume rm testVolXXXXX]: exit status 1
2017/06/08 18:37:08 Sleep for 2 seconds
2017/06/08 18:37:10 Volume cannot be deleted yet as plugin initialization still in progress. Retrying...
2017/06/08 18:37:11 Successfully deleted volume testVolXXXXX
```

### CODE REVIEW

File: tests/e2e/vmlistener_test.go [hunk 1]
```diff
@@ -16,8 +16,6 @@
 // In case of vm/esx host restart vDVS cleans up stale attachment details
 // the volume.
 
-// +build unstable
```
Comment (shuklanirdesh82) [inline tests/e2e/vmlistener_test.go]: why? we do need tag here.

File: tests/e2e/vmlistener_test.go [hunk 2]
```diff
@@ -128,5 +126,5 @@ func (s *VMListenerTestParams) TestKillVM(c *C) {
 	status = verification.VerifyDetachedStatus(s.volumeName, s.dockerHostIP, s.esxIP)
 	c.Assert(status, Equals, true, Commentf("Volume %s is still attached", s.volumeName))
 
-	log.Printf("END: Test vmdkops service restart + kill vm process")
+	log.Printf("\n END: Test vmdkops service restart + kill vm process")
```
Comment (shuklanirdesh82) [inline tests/e2e/vmlistener_test.go]: let's bring this one to the latest with utils from misc

File: tests/utils/dockercli/volumelifecycle.go [hunk 3]
```diff
@@ -85,7 +90,21 @@ func ReadFromVolume(ip, volName, containerName, fileName string) (string, error)
 // DeleteVolume helper deletes the created volume as per passed volume name.
 func DeleteVolume(ip, name string) (string, error) {
 	log.Printf("Destroying volume [%s]\n", name)
-	return ssh.InvokeCommand(ip, dockercli.RemoveVolume+name)
+	var out string
+	var err error
+	for attempt := 0; attempt <= maxAttempt; attempt++ {
```
Comment (shuklanirdesh82) [inline tests/utils/dockercli/volumelifecycle.go]: if you are planning to set 30seconds max then suppose to start with `attempt=1` or `attempt < maxAttempt`

File: tests/utils/dockercli/volumelifecycle.go [hunk 4]
```diff
@@ -85,7 +90,21 @@ func ReadFromVolume(ip, volName, containerName, fileName string) (string, error)
...
+		out, err = ssh.InvokeCommand(ip, dockercli.RemoveVolume+name)
+		if err != nil {
+			if strings.Contains(out, "Plugin initialization in progress.") {
```
Comment (shuklanirdesh82) [inline tests/utils/dockercli/volumelifecycle.go]: move this as string const..would be easier if message changes in the future.

File: tests/utils/dockercli/volumelifecycle.go (hunk 3 above)
Comment (govint) [inline tests/utils/dockercli/volumelifecycle.go]: 0 - (maxAttempt - 1)

File: tests/utils/dockercli/volumelifecycle.go [hunk 5]
```diff
@@ -85,7 +90,21 @@ func ReadFromVolume(ip, volName, containerName, fileName string) (string, error)
...
+				misc.SleepForSec(waitTime)
+				log.Printf("Volume cannot be deleted yet as plugin initialization still in progress. Retrying...")
+				continue
+			}
+			return out, err
+		}
+		break
+	}
+	return out, err
```
Comment (pshahzeb) [inline tests/utils/dockercli/volumelifecycle.go]: Can avoid nesting.
You only need to wait when error is not nil (exit status 1) and output message contains the error string

```
	var out string
	var err error

	for attempt := 0; attempt <= maxAttempt; attempt++ {
		out, err := ssh.InvokeCommand(ip, dockercli.RemoveVolume+name)
		if err != nil && strings.Contains(out, "Plugin initialization in progress.") {
			misc.SleepForSec(waitTime)
			log.Printf("Volume cannot be deleted yet as plugin initialization still in progress. Retrying...")
			continue
		} else {
			break
		}
	}

	return out, err
```

File: tests/utils/dockercli/volumelifecycle.go [hunk 6]
```diff
@@ -27,6 +27,11 @@ import (
 	"github.com/vmware/docker-volume-vsphere/tests/utils/ssh"
 )
 
+const (
+	maxAttempt = 15
```
Comment (pshahzeb) [inline tests/utils/dockercli/volumelifecycle.go]: maxRemoveVolAttempt

File: tests/e2e/vmlistener_test.go (hunk 1 above)
Comment (shuklanirdesh82) [inline tests/e2e/vmlistener_test.go]: `runalways ` => `runonce`

**Pull request merged**